sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -555,30 +555,49 @@ class RocksDB(
 
   private def loadWithoutCheckpointId(
       version: Long,
-      readOnly: Boolean = false): RocksDB = {
+      readOnly: Boolean = false,
+      createEmpty: Boolean = false): RocksDB = {
+
     try {
-      if (loadedVersion != version) {
+      enableChangelogCheckpointing = if (createEmpty) false else conf.enableChangelogCheckpointing
```

Comment:
Do this in the `load` method since we will also need this for the loadwithcheckpointId func later. Also add comment why we are doing this.